components/dashboard/src/projects/Prebuilds.tsx

```diff
@@ -150,7 +165,7 @@ export default function () {
                 <div className="py-3 pl-3">
                     <DropDown prefix="Prebuild Status: " contextMenuWidth="w-32" entries={statusFilterEntries()} />
                 </div>
-                <button disabled={!project} onClick={() => triggerPrebuild(null)} className="ml-2">Trigger Prebuild</button>
```

Comment:
**suggestion**: Regarding the removal of the global _Trigger Prebuild_ button mentioned in https://github.com/gitpod-io/gitpod/pull/5731#issuecomment-923857047 (Cc @svenefftinge), the linked issue (https://github.com/gitpod-io/gitpod/issues/5374) in the PR description contains some more context behind this:

From https://github.com/gitpod-io/gitpod/issues/5374#issuecomment-905420796:
> In terms of UX, this button could trigger a modal to confirm prebuild on the default branch. Possible improvements could include selecting a different branch, passing an env var for a prebuild, etc.

From https://github.com/gitpod-io/gitpod/issues/5374#issuecomment-911410680:
> **We could skip this and remove the button for this iteration if you think it’s redundant.** The idea behind this is that for projects that the latest prebuild failed and you’d like to retry or all the prebuilds have been removed after two weeks, someone could trigger a prebuild from the dashboard.

> However, if we want to ship this sooner and we have the bandwidth to add a fix here, showing a spinner + disabling the button on click as you mentioned in #5374 (comment) sounds optimal. 🎯

> I refrained from suggesting this earlier as there are also a few more places where a loading button component variant could be used for consistency, like the git integration modal, but we can progressively add it there later. ➿

🍊 🍊 🍊 🍊 

Adding below some early design drafts how the confirmation modal could look like when triggering a prebuild from the global trigger prebuild button. Let me know if this does not look interesting or useful here. Using a modal here could allow us in the future to provide a way to insert one-time variables for a prebuild which, etc.

| Run Prebuild | Run Prebuild with Branch Selection |
|-|-|
| <img width="1440" alt="run-1" src="https://user-images.githubusercontent.com/120486/134175794-303b4f69-667e-4b31-a64a-f6628ff7cdeb.png"> | <img width="1440" alt="run-2" src="https://user-images.githubusercontent.com/120486/134175787-27d4ad0f-3dda-4ff6-9fe2-587013008618.png"> |

🍋 🍋 🍋 🍋 

Regarding the replacement of an existing prebuild upon triggering a new prebuild mentioned in https://github.com/gitpod-io/gitpod/pull/5731#issuecomment-923861134 (Cc @jankeromnes), I'd think that ideally we'd never replace a prebuild but cancel the previous one and creating a new one. Maybe this is currently not technically possible or something we're interested in? 💭